core/iwasm/interpreter/wasm_interp_classic.c

```diff
@@ -35,15 +35,21 @@ typedef float64 CellType_F64;
 
 #define BR_TABLE_TMP_BUF_LEN 32
 
-#if WASM_ENABLE_THREAD_MGR == 0
-#define get_linear_mem_size() linear_mem_size
-#else
+#if WASM_ENABLE_THREAD_MGR != 0
 /**
  * Load memory data size in each time boundary check in
  * multi-threading mode since it may be changed by other
  * threads in memory.grow
  */
 #define get_linear_mem_size() GET_LINEAR_MEMORY_SIZE(memory)
+#elif WASM_ENABLE_MULTI_MEMORY != 0
+/**
+ * Load memory data size in each time boundary check in
+ * multi-memory mode since the memory instance(memories[i]) may be different
+ */
```

Comment:
alternatively, you can update `linear_mem_size` (and `is_memory64` in future) when updating `memory`.

Reply:
good idea, I will try that